dds/DCPS/NetworkConfigModifier.h

```diff
@@ -10,6 +10,9 @@
 
 #include "ace/config.h"
 
+// ACE_HAS_GETIFADDRS is not set on android but is available in API >= 24
+#if ((!defined (ACE_LINUX) && defined(ACE_HAS_GETIFADDRS))  || (defined(ACE_ANDROID) && __ANDROID_API__ >= 24)) && !defined(OPENDDS_SAFETY_PROFILE)
```

Comment:
So I think this is almost right. I would say `(defined(ACE_ANDROID) && !defined ACE_LACKS_IF_NAMEINDEX)` because [ACE 6's condition is different](https://github.com/DOCGroup/ACE_TAO/blob/f1733f33df42f11ceade9575a010cc5a0495a8af/ACE/ace/config-android.h#L137) from the ACE master one I accidentally linked before.